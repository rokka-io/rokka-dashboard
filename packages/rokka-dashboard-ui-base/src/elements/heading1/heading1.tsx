import styled from 'styled-components';
import { fonts } from '../../identity/typography/index';

interface Heading1StylingProps {
  /** Color of heading text */
  color?: string;
}

export const Heading1 = styled.h1<Heading1StylingProps>`
  font-family: ${fonts.families.brand};
  font-size: ${fonts.sizes.xlarge};
  color: ${({ color }) => color};
`;
