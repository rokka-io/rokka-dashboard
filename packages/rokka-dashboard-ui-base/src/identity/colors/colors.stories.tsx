import React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import { colors } from './colors';

function generateColorList(key: string): JSX.Element[] {
  return Object.keys(colors[key]).map(name => (
    <Swatch key={name}>
      <Color bg={colors[key][name]} />
      <Values>
        <ColorName>{name}</ColorName>
        <ColorValue>{colors[key][name]}</ColorValue>
      </Values>
    </Swatch>
  ));
}

storiesOf('Identities / Colors', module)
  .add('Brand', () => generateColorList('brand'))
  .add('Gray', () => generateColorList('gray'))
  .add('Cranberry', () => generateColorList('cranberry'))
  .add('Orange', () => generateColorList('orange'))
  .add('Tints', () => generateColorList('tints'));

const Swatch = styled.li`
  border: none;
  border-radius: 0.25rem;
  list-style: none;
  display: inline-flex;
  flex-direction: column;
  margin: 0 0.5rem 2rem;
  background: #efefef;
  box-shadow: 0 8px 6px -6px rgba(0, 0, 0, 0.2);
`;

const Color = styled.output<{ bg: string }>`
  background: ${({ bg }) => bg};
  width: 10rem;
  height: 6rem;
  border-radius: 0.25rem 0.25rem 0 0;
  border: 1px solid black;
`;

const ColorName = styled.var`
  font-style: normal;
  text-transform: uppercase;
  font-size: 0.8rem;
  color: #898989;
  display: block;
  margin-bottom: 1rem;
`;

const ColorValue = styled.var`
  font-style: normal;
  text-transform: uppercase;
  font-size: 1.2rem;
  color: #444;
  display: block;
`;

const Values = styled.div`
  text-align: center;
  display: inline-block;
  padding: 1rem 0;
`;
