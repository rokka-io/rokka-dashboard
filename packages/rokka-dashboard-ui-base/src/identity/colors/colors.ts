export const colors: { [index: string]: { [index: string]: string } } = {
  brand: {
    dark: '#4d852c',
    light: '#8DB547',
    lightest: '#A4C339',
    primary: '#6EA644',
    avatar: '#588536'
  },
  cranberry: {
    primary: '#D34361'
  },
  gray: {
    dark: '#C0C1BC',
    darker: '#A5A59F',
    darkest: '#898A81',
    light: '#DDDDD8',
    lighter: '#EBEBE6',
    lightest: '#F7F7F5',
    primary: '#CECFCA'
  },
  orange: {
    dark: '#ec9506'
  },
  tints: {
    black: '#000',
    white: '#fff'
  }
};
