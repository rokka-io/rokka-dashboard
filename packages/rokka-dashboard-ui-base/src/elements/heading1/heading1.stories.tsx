import { color, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';

import { colors } from '../../identity/colors/colors';
import { Heading1 } from './heading1';

storiesOf('Elements / Headings', module).add('Heading 1', () => (
  <Heading1 color={color('Color', colors.tints.black)}>{text('text', 'A Heading')}</Heading1>
));
