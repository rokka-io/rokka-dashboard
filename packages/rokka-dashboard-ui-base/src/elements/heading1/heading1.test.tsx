import { shallow } from 'enzyme';
import React from 'react';

import { colors } from '../../identity/colors/colors';
import { Heading1 } from './heading1';

describe('Elements', () => {
  describe('Heading1', () => {
    it('renders a heading 1 correctly.', () => {
      const component = shallow(<Heading1 color={colors.tints.black}>Heading</Heading1>);

      expect(component).toMatchSnapshot();
    });
  });
});
