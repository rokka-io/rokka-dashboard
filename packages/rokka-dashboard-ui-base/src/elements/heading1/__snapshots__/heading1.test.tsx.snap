// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements Heading1 renders a heading 1 correctly. 1`] = `
<StyledComponent
  color="#000"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-family: ",
          "\\"Liip Etica Book\\", sans-serif",
          ";
  font-size: ",
          "1.5rem",
          ";
  color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.h1",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "h1",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Heading
</StyledComponent>
`;
